rubble/src/link/advertising.rs

```diff
@@ -570,17 +570,15 @@ impl PduBuf {
     /// Creates an advertising channel PDU suitable for building a simple
     /// beacon.
     ///
-    /// This is mostly equivalent to `PduBuf::nonconnectable_undirected`, but it
-    /// will automatically add a suitable `Flags` AD structure to the
-    /// advertising data (this flags is mandatory).
+    /// This is equivalent to `PduBuf::nonconnectable_undirected`.
     pub fn beacon(
         advertiser_addr: DeviceAddress,
         advertiser_data: &[AdStructure<'_>],
     ) -> Result<Self, Error> {
         Self::adv(
             PduType::AdvNonconnInd,
             advertiser_addr,
-            &mut iter::once(&AdStructure::from(Flags::broadcast())).chain(advertiser_data),
+            &mut advertiser_data.iter(),
         )
```

Comment:
The body can just forward to `nonconnectable_undirected` now

Reply:
I'll change it

Reply:
done